revolv/static/js/base/dashboard.js

```diff
@@ -20,6 +20,73 @@ $(document).ready(function () {
             $this.addClass("fa-rotate-180");
         }
     });
+    
+    /**
+     * This function defines what happens when the different categories are clicked in the
+     * 'my impact' section on the dashboard. The category that is clicked is made active, 
+     * and then an AJAX request is made. The view this AJAX reqest is sent to updates the
+     * preferred_categories attribute of this user.
+     * 
+     */
+    $(".category-option-container").click(function() {
+        $(this).toggleClass("active");
+        var allActiveCategories = $(".category-option-container.active");
+        var allCategories = $(".category-option-container");
+        var dict = {};
+        for (var i = 0; i < allCategories.length; i += 1) {
+            var categoryText = $(allCategories[i]).children().last().text();
+            dict[categoryText] = false;
+        }
+
+        for (var i = 0; i < allActiveCategories.length; i += 1) {
+            var categoryText = $(allActiveCategories[i]).children().last().text();
+            dict[categoryText] = true;
+        }
+        makeAJAXCall(dict);
+    })
+
+    /**
+     * Makes an AJAX Call.
+     * @param {Object} changed_data - An object with either 0 or 1 keys that maps a selector to its new value.
+     */
+    function makeAJAXCall(changedData) {
+        var csrftoken = getCookie('csrftoken');
+        $.ajax({
+            url: window.categoryURL,
+            method: "GET",
+            data: changedData,
+            headers: {"X-CSRFToken": csrftoken},
+        });
+    }
+
+    /**
+     * Copied from the Django documentation. Checks if a method needs a csrf token.
```

Comment:
Can you include the url of the django documentation page where this came from? I've used this before too haha, but it's always nice to have a url for someone who hasn't to better be able to understand